Fix Format-Table where rows were being trimmed unnecessarily if there's only one row of headers

SteveL-MSFT opened the pull request:
## PR Summary

In the case where a row doesn't expand to multiple rows, columns were still being trimmed of whitespace which introduces invalid alignment as the whitespace needs to be retained.
Added new tests to cover this scenario.  Updated some existing tests to use `^` instead of backtick to represent newline making it easier to replace it back with a newline to visualize during debugging when test fails.

Fix https://github.com/PowerShell/PowerShell/issues/6767

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

SteveL-MSFT committed:
add test where value in table is wider than header, but wrap is false

SteveL-MSFT committed:
[feature]
don't trim columns in table if there's only one row
added tests

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Table.Tests.ps1:
```diff
@@ -668,6 +678,40 @@ er
+
+abc  bcd
+---  ---
+1234 123
```
Should 4 be on new next line?
```
abc bcd
--- ---
123 123
4

SteveL-MSFT replied:
That's only if there is a format file that specifies the width.  When there isn't it will auto set the width of a column based on the first few samples of the output.  So in this case, the `1234` dictate the width of column 1.

iSazonov replied:
Is the fix (and the bug) for autosize only?

iSazonov replied:
Could you please add a comment or change the test header that it is a _implicitly autosize_?
(Also we should add later tests for explicit `-AutoSize`.)

SteveL-MSFT replied:
Updated test description to make clear and added more rows to cover accidental trim of a row and not just header

SteveL-MSFT committed:
[feature]
address Ilya's feedback and added test to capture rows that got trimmed and not just header
